src/Backend/Teradata/Helper/BackendHelper.php

```diff
@@ -15,4 +17,51 @@ public static function generateTempDedupTableName(): string
     {
         return '__temp_DEDUP_' . str_replace('.', '_', uniqid('csvimport', true));
     }
+
+    /**
+     * creates a wildcard string which should match all files in manifest
+     * [file01.csv, file01.csv] => file0*
+     * TODO
+     *  - has to fix edgecases a) [1_file.csv, 2_file.csv] b) not all the files matched in WC have to be on s3
+     * @param SourceFile $source
+     * @return string
+     * @throws \Keboola\Db\Import\Exception
+     */
+    public static function getMask(SourceFile $source): string
```

Comment:
Tohle jde (IMHO) mnohem efektivněji udělat tak, že vezmeš první věc co najdeš a odzadu z ní odkrajuješ. Budeš mít výsledek dřív a současně je to čitelnější. 

mám:
$strings = [
transformator001
transformator00
transformator101
]

Vezmu první:
```
$search = transformator001;
found = false;
while (!found) {
found = true;
foreach $strings  as $string
 if (strpos($search, $string) !== 0) {
  $search = transformator00 (smazán poslední znak)
  $found = false;
  break;
 }
}
```

Reply:
Potenciálně můžeš mít velmi dlouhé stringy a může jich být tisíce, takže tady se zrovna optimalizace vyplatí IMHO...

Reply:
pocitam s tim, ze je to docasne reseni. Je na to storka tady https://keboola.atlassian.net/browse/KBC-2533 . Ono to ma imho dost der - uplne jiny nazev souboru, cislo na zacatku (001_file) apod.

Reply:
Kdyby to bylo jen o výkonu tak neřeknu nic, ale IMHO je to nečitelné a obtížně pochopitelné co to dělá :) Ale OK

Reply:
![image](https://user-images.githubusercontent.com/642928/159708686-02df5485-868b-44da-8657-ac2ce6df8090.png)

Reply:
nerikal jsi, ze jdes s detma ven ? :)

Reply:
No je to dobrý hegeš ten kód. Chvílu sem to četl, než sem to pochopil, doplnil sem tam test, ten cos tam měl byl taky značně na prd a testoval to na něčem co sa v manifestu ani neoběví. Od zadu to podle mě neuděláš, protože sů tam soubory třeb `tw_accounts.csv.gz*_part_00.gz` aspoň v testech, úplně nevím co tam všecko može nakadit komponenta za slicy, ale aspoň exporty z connection dávajů pěkné manifesty se stabilníma názvama :)

Reply:
Každopádně je to na hovno, že něco takového potřebujeme, protože to je všecko in memory, a už máme tu cool issue, že sa nevleze import do paměti když je v manifestu 300k položek :D